resolver-dns/src/main/java/io/netty/resolver/dns/DnsAddressResolverGroup.java

```diff
@@ -42,23 +42,27 @@
 @UnstableApi
 public class DnsAddressResolverGroup extends AddressResolverGroup<InetSocketAddress> {
 
-    private final ChannelFactory<? extends DatagramChannel> channelFactory;
-    private final DnsServerAddressStreamProvider nameServerProvider;
+    private final DnsNameResolverBuilder dnsResolverBuilder;
 
     private final ConcurrentMap<String, Promise<InetAddress>> resolvesInProgress = newConcurrentHashMap();
     private final ConcurrentMap<String, Promise<List<InetAddress>>> resolveAllsInProgress = newConcurrentHashMap();
 
+    public DnsAddressResolverGroup(DnsNameResolverBuilder dnsResolverBuilder) {
+        this.dnsResolverBuilder = dnsResolverBuilder;
```

Comment:
This won't work.   If someone makes a DnsNameResolverBuilder and then calls build, and then subsequently modifies the builder, the channelFactory of DnsAddressResolverGroup will change. 

That would be unexpected.

Reply:
@carl-mastrangelo Not sure I follow. If you passed `DnsNameResolverBuilder` to create a `DnsAddressResolverGroup`, what you would expect from changing `channelFactory` on the builder you've used?

Reply:
I wouldn't expect the DnsAddressResolverGroup to use a different channel factory if the DnsNameResolverBuilder changed channelFactory.    That is currently possible as is.

Reply:
@carl-mastrangelo Would a deep copy of the builder meet your expectations? This will prevent mutations propagation, but it might be not expected as well.

Reply:
a deep copy would be good.   Immutability is highly recommended.  This is Effective Java item 15 in the the 2nd edition.

Reply:
@carl-mastrangelo Updated.